### PR TITLE
Issue 1787 fixing connector endpoint returns index not found

### PR DESCRIPTION
**Pull request by samuel-oci**
### Description
When connector is not created yet, return an empty search response instead of a broken 404 with cryptic index not found message.
More description in: https://github.com/opensearch-project/ml-commons/issues/1787

Minor additional issues fixed:
- Ability to run integTest with cluster debugging. 
For example:
`./gradlew :opensearch-ml-plugin:integTest --tests org.opensearch.ml.rest.RestMLSearchConnectorActionIT.testSearchConnector_beforeConnectorCreated -Dcluster.debug=1`
- Show log output during integTest

### Issues Resolved
[[List any issues this PR will resolve]](https://github.com/opensearch-project/ml-commons/issues/1787)
 
### Check List
- [ x] New functionality includes testing.
  - [x ] All tests pass
- [x ] New functionality has been documented.
  - [x ] New functionality has javadoc added
- [ x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java [hunk 1]
```diff
@@ -78,12 +81,38 @@ private void search(SearchRequest request, ActionListener<SearchResponse> action
                 excludes.toArray(new String[0])
             );
             request.source().fetchSource(rebuiltFetchSourceContext);
+
+            ActionListener<SearchResponse> doubleWrappedListener = ActionListener.wrap(wrappedListener::onResponse, e -> {
+                if (e instanceof IndexNotFoundException) {
```
Comment (Hailong-am) [inline plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java:86]: better to use `ExceptionHelp.unwrap` to unwrap the exception for multiple node case.

File: plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java [hunk 2]
```diff
@@ -78,12 +81,38 @@ private void search(SearchRequest request, ActionListener<SearchResponse> action
...
+                        0,
+                        0,
+                        0,
+                        0,
+                        null,
+                        new ShardSearchFailure[] {},
+                        SearchResponse.Clusters.EMPTY,
+                        null
+                    );
+                    wrappedListener.onResponse(emptySearchResponse);
```
Comment (Hailong-am) [inline plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java:104]: return empty result with status 200 may confusing user, better to keep rest status as 404 with customized error message.

Comment (samuel-oci) [reply plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java:104]: > return empty result with status 200 may confusing user, better to keep rest status as 404 with customized error message.

Why should user get or expect to get 404 if all they did was to ask which connectors are available?
The return of empty result is inline with the expected behavior described in the issue https://github.com/opensearch-project/ml-commons/issues/1787

File: plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java (hunk 1 above)
Comment (Hailong-am) [reply plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java:86]: just update to `unwrapCause`

Comment (samuel-oci) [reply plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java:86]: Sounds good, will do

Comment (samuel-oci) [reply plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java]: @Hailong-am done

**Comment by ylwu-amzn** [comment]
The code looks good to me. How about enhancing other search APIs too:  search model / model group/ tasks?

**Comment by samuel-oci** [comment]
> The code looks good to me. How about enhancing other search APIs too: search model / model group/ tasks?

@ylwu-amzn sounds good, would you like me to create issues for those as well?

File: plugin/build.gradle [hunk 3]
```diff
@@ -198,7 +202,7 @@ testClusters.integTest {
     // When running integration tests it doesn't forward the --debug-jvm to the cluster anymore
     // i.e. we have to use a custom property to flag when we want to debug elasticsearch JVM
     // since we also support multi node integration tests we increase debugPort per node
-    if (System.getProperty("opensearch.debug") != null) {
+    if (System.getProperty("cluster.debug") != null) {
```
Comment (Hailong-am) [inline plugin/build.gradle:205]: have you tried with option `-debug-jvm` that should work also.

**Review by Hailong-am — APPROVED** [review]

File: plugin/build.gradle (hunk 3 above)
Comment (samuel-oci) [reply plugin/build.gradle:205]: Yeah, I did try that. But unfortunately it doesn't seem to be passed to the actual cluster being spun by the integ test.
So the test itself is debuggable but not the cluster spun by it.

**Review by HenryL27 — APPROVED** [review]

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**